File: app/src/main/java/com/supercharge/Homework/service/MovieService.java
package com.supercharge.Homework.service;

import com.supercharge.Homework.model.response.MoviesResponseBO;
import retrofit2.Call;
import retrofit2.http.Field;
import retrofit2.http.FormUrlEncoded;
import retrofit2.http.GET;
import retrofit2.http.Query;

public interface MovieService {

    @GET("/3/search/movie")
    Call<MoviesResponseBO> getAllMoviews(
            @Query("api_key") String apiKey,
            @Query("language") String language,
            @Query("query") String query
    );

    @GET("/3/movie")
    Call<MoviesResponseBO> getMovieById(
            @Query("api_key") String apiKey,
            @Query("language") String language
    );
}

File: app/src/main/java/com/supercharge/Homework/service/MovieService.kt
package com.supercharge.Homework.service

import com.supercharge.Homework.model.MovieBO
import com.supercharge.Homework.model.response.MoviesResponseBO
import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Path
import retrofit2.http.Query

interface MovieService {
    @GET("/3/search/movie")
    fun getAllMoviews(
        @Query("api_key") apiKey: String,
        @Query("language") language: String,
        @Query("query") query: String
    ): Call<MoviesResponseBO>

    @GET("/3/movie/{movie_id}")
    fun getMovieById(
        @Path(value = "movie_id", encoded = true) movieId: String,
        @Query("api_key") apiKey: String,
        @Query("language") language: String
    ): Call<MovieBO>
}